Document Vue (#634)
parcel-bundler/website: src/languages/vue.md

---
layout: layout.njk
eleventyNavigation:
  key: languages-vue
  title: <img src="/assets/lang-icons/vue.svg" alt=""/> Vue
  order: 4
---

> Note that Parcel 2 _does not support Vue 2_ if you plan to use SFCs; you must use [Vue 3 beta](https://github.com/vuejs/vue-next) or later.

Reviewer: ```suggestion
Note that Parcel does not support using SFCs with Vue 2, you must use [Vue 3 beta](https://github.com/vuejs/vue-next) or later.
```

Vue.js is a progressive, incrementally-adoptable JavaScript framework for building UI on the web. Parcel supports Vue without the need for any additional configuration.

{% sample %}
{% samplefile "index.html" %}

```html
<div id="app"></div>
<script src="./index.js"></script>
```

{% endsamplefile %}
{% samplefile "index.js" %}

```jsx
import { createApp } from "vue";
import App from "./App";

const app = createApp(App);
app.mount("#app");
```

{% endsamplefile %}
{% samplefile "App.vue" %}

```html
<template>
  <div>Hello {% raw %}{{ name }}{% endraw %}!</div>
</template>

<script>
  export default {
    data() {
      return {
        name: "Vue",
      };
    },
  };
</script>
```

{% endsamplefile %}
{% endsample %}

## HMR

> Parcel 2 uses the same HMR injection as the official [vue-loader](https://github.com/vuejs/vue-loader) for Webpack, so you'll have a fast, reactive development experience.

Reviewer: ```suggestion
Parcel uses the same HMR injection as the official [vue-loader](https://github.com/vuejs/vue-loader) for Webpack, so you'll have a fast, reactive development experience.
```
